generators/php/base/src/asIs/Client/RawClient.Template.php

```diff
@@ -87,60 +97,110 @@ private function buildRequest(
             MultipartApiRequest::class => array_merge(
                 $this->headers,
-                $request->headers
+                $request->headers,
+                $options['headers'] ?? [],
             ),
             default => throw new InvalidArgumentException('Unsupported request type: ' . get_class($request)),
         };
     }
 
+    /**
+     * @param BaseApiRequest $request
+     * @param array{
+     *     bodyProperties?: array<string, mixed>,
+     * } $options
+     * @return ?StreamInterface
+     */
     private function encodeRequestBody(
-        BaseApiRequest $request
-    ): ?StreamInterface
-    {
+        BaseApiRequest $request,
+        array $options,
+    ): ?StreamInterface {
         return match (get_class($request)) {
-            JsonApiRequest::class => $request->body != null ? Utils::streamFor(json_encode($request->body)) : null,
+            JsonApiRequest::class => $request->body === null ? null : Utils::streamFor(
+                json_encode(
+                    $this->buildJsonBody(
+                        $request->body,
+                        $options,
+                    ),
+                )
+            ),
             MultipartApiRequest::class => $request->body != null ? new MultipartStream($request->body->toArray()) : null,
             default => throw new InvalidArgumentException('Unsupported request type: ' . get_class($request)),
         };
     }
 
+    /**
+     * @param mixed $body
+     * @param array{
+     *     bodyProperties?: array<string, mixed>,
+     * } $options
+     * @return mixed
+     */
+    private function buildJsonBody(
+        mixed $body,
+        array $options,
+    ): mixed {
+        $overrideProperties = $options['bodyProperties'] ?? [];
+        if ($body instanceof JsonSerializable) {
+            $serialized = $body->jsonSerialize();
+            return is_array($serialized)
+                ? array_merge($serialized, $overrideProperties)
+                : $serialized;
+        }
+        return is_array($body)
+            ? array_merge($body, $overrideProperties)
+            : $body;
+    }
+
+    /**
+     * @param BaseApiRequest $request
+     * @param array{
+     *     queryParameters?: array<string, mixed>,
+     * } $options
+     * @return string
+     */
     private function buildUrl(
-        BaseApiRequest $request
-    ): string
-    {
+        BaseApiRequest $request,
+        array $options,
+    ): string {
         $baseUrl = $request->baseUrl;
         $trimmedBaseUrl = rtrim($baseUrl, '/');
         $trimmedBasePath = ltrim($request->path, '/');
         $url = "{$trimmedBaseUrl}/{$trimmedBasePath}";
-
-        if (!empty($request->query)) {
-            $url .= '?' . $this->encodeQuery($request->query);
+        $query = array_merge(
+            $request->query,
+            $options['queryParameters'] ?? [],
+        );
+        if (!empty($query)) {
+            $url .= '?' . $this->encodeQuery($query);
         }
```

Comment:
Instead of adding this to the URL ourselves, can we use the built-in query option?
We should be mindful of how this is passed to the request method, which may override or merge the query values from the client/request/request method.

https://docs.guzzlephp.org/en/stable/request-options.html#query

Reply:
Nice find + good call - I think it's best to update to use the built-in query option in a follow-up so that we have a commit focused on that potential behavioral change. The query parameters aren't passed in as options now, so we aren't susceptible to those side effects in the meantime.